### DISCUSSION TITLE
Add ability to select CPU's and memory when creating dev environment?

### DISCUSSION THREAD

**Discussion by argia-andreas** [Vessel-App/vessel-cli#19] [Ideas]
Would you be open for a PR that adds ability to select amount of cpus and memory(256, 512, 1024, 2048) on running vessel init?
With the lowest being the default.

Like so:
<img width="635" alt="image" src="https://user-images.githubusercontent.com/9297340/190896043-e5397569-c0bd-4450-890f-9b74da0a937a.png">

Had issues running npm install and npm run build due to low memory.

Setting more memory when creating the environment solved it for me.

**Comment by fideloper** [comment]
I do like that but I think it may need to be a separate command instead of part of the init command.

The reason I’m thinking that is that upgrading the server size moves you out of the free tier  and can fail validation on Fly’s end when trying to start a VM before adding a payment method.

**Reply by fideloper** [reply]
It occurs to me that I need to double check that the machines API is available to accounts with no payment method. I did my testing with a blessed employee account!

**Reply by argia-andreas** [reply]
Yeah..
Tested with a freshly registered account:
<img width="1280" alt="image" src="https://user-images.githubusercontent.com/9297340/190954820-595e20ea-bc89-4a84-bc69-e42217dfde8c.png">


I added a MySql instance so had to add payment information pretty quickly so didn't notice the failure att the beginning.

**Reply by argia-andreas** [reply]
So for making upgrading/upsizing the server size separate from creating a dev environment, 
`vessel scale` or `vessel size` perhaps?

And you get to choose machine type(shared-cpu, dedicated-cpu), CPU's and memory?

Or you could also just show a list
(shared-cpu-1x, dedicated-cpu-1x, dedicated-cpu-2x, and so on)
And select memory after that.

**Reply by fideloper** [reply]
This is an interesting one! If you check the pricing page, it has the best info on the sizes and how you can scale memory on them: https://fly.io/docs/about/pricing/

Ideally this stays simple and just  allows the named sizes (giving them the minimum amount of ram). At the same time, I think it would be good if the shared 1cpu was allowed to scale memory up without going into the dedicated range (since they're the cheapest and likely most desired).

**So maybe the options are:**

* `shared-cpu-1x - 256m`
* `shared-cpu-1x - 512m`
* `shared-cpu-1x - 1g`
* `shared-cpu-1x - 2g`
* `dedicated-cpu-1x - 2g`
* `dedicated-cpu-2x -4g`
* `dedicated-cpu-4x - 8g`
* `dedicated-cpu-8x - 16g`

And perhaps the output asking could even link to the pricing page so people can see what they're getting into (most terminals lets you click or cmd+click links to open them in the browser).

> I wouldn't adding specific prices to the output since it's subject to change.

**Last note:** I personally like the `vessel scale` command the best.

**Reply by argia-andreas** [reply]
Sounds good!
I'll get started on a PR :-)
